app/src/main/java/fr/groggy/racecontrol/tv/ui/season/archive/SeasonArchiveFragment.kt

```diff
@@ -13,28 +13,43 @@ import fr.groggy.racecontrol.tv.ui.season.browse.SeasonBrowseActivity
 
 @Keep
 @AndroidEntryPoint
-class SeasonArchiveFragment: VerticalGridSupportFragment(), OnItemViewClickedListener {
-    private val itemAdapter: ArrayObjectAdapter by lazy {
-        ArrayObjectAdapter(ArchivePresenter())
-    }
+class SeasonArchiveFragment: BrowseSupportFragment(), OnItemViewClickedListener {
+
+    private val archivesAdapter = ArrayObjectAdapter(ListRowPresenter())
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setupUIElements()
+        setupEventListeners()
+        buildRowsAdapter()
+    }
 
+    private fun setupUIElements() {
         title = getText(R.string.choose_a_season)
-        gridPresenter = VerticalGridPresenter().apply {
-            numberOfColumns = 5
-        }
-        adapter = itemAdapter
-        onItemViewClickedListener = this
+        headersState = HEADERS_ENABLED
+        isHeadersTransitionOnBackEnabled = true
+        brandColor = ContextCompat.getColor(requireContext(), R.color.fastlane_background)
+        adapter = archivesAdapter
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
+    private fun setupEventListeners() {
+        onItemViewClickedListener = this
+    }
 
+    private fun buildRowsAdapter() {
+        val archivesTitle = "%s - %s"
```

Comment:
Can you move this to the strings resource file? As this is a UI related stuff, if for some reason we decide to add anything on future it is a single place to change